src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1265,81 +1268,176 @@ public void ForceRun()
+          // was moved from the ViewModel. When #4258 is implemented, we will need to
+          // remove this step.
+          var currentHomeSpaces = Workspaces.OfType<HomeWorkspaceModel>().ToList();
+          if (currentHomeSpaces.Any())
+          {
+            // If the workspace we're opening is a home workspace,
+            // then remove all the other home workspaces. Otherwise,
+            // Remove all but the first home workspace.
+            var end = ws is HomeWorkspaceModel ? 0 : 1;
 
-                        OnWorkspaceOpening(xmlDoc);
+            for (var i = currentHomeSpaces.Count - 1; i >= end; i--)
+            {
+              RemoveWorkspace(currentHomeSpaces[i]);
+            }
+          }
 
-                        // TODO: #4258
-                        // The following logic to start periodic evaluation will need to be moved
-                        // inside of the HomeWorkspaceModel's constructor.  It cannot be there today
-                        // as it causes an immediate crash due to the above ResetEngine call.
-                        var hws = ws as HomeWorkspaceModel;
-                        if (hws != null)
-                        {
-                            // TODO: #4258
-                            // Remove this ResetEngine call when multiple home workspaces is supported.
-                            // This call formerly lived in DynamoViewModel
-                            ResetEngine();
+          AddWorkspace(ws);
+          CurrentWorkspace = ws;
+        }
 
-                            if (hws.RunSettings.RunType == RunType.Periodic)
-                            {
-                                hws.StartPeriodicEvaluation();
-                            }
-                        }
+        private void SetPeriodicEvaluation(WorkspaceModel ws)
+        {
+          // TODO: #4258
+          // The following logic to start periodic evaluation will need to be moved
+          // inside of the HomeWorkspaceModel's constructor.  It cannot be there today
+          // as it causes an immediate crash due to the above ResetEngine call.
+          var hws = ws as HomeWorkspaceModel;
+          if (hws != null)
+          {
+            // TODO: #4258
+            // Remove this ResetEngine call when multiple home workspaces is supported.
+            // This call formerly lived in DynamoViewModel
+            ResetEngine();
 
-                        CurrentWorkspace = ws;
-                        return;
-                    }
-                }
+            if (hws.RunSettings.RunType == RunType.Periodic)
+            {
+              hws.StartPeriodicEvaluation();
             }
-            Logger.LogError("Could not open workspace at: " + xmlPath);
+          }
+        }
+
+        private bool OpenJsonFile(WorkspaceInfo workspaceInfo, string fileContents, out WorkspaceModel workspace)
+        {
+            CustomNodeManager.AddUninitializedCustomNodesInPath(
```

Comment:
has this been tested with some dyf files as JSON, do they actually load? I'm not sure we've thought all that through yet.